### DISCUSSION TITLE
Does this lib solves the problem when turbo frames are loaded asynchronously?

### DISCUSSION THREAD

**Discussion by luizkowalski** [SimoTod/alpine-turbo-drive-adapter#56] [Q&A]
hey! first of all, thanks for this project!

I wanted to ask you if this gem solves the problem described in [this thread](https://discuss.hotwired.dev/t/issue-with-hotwire-apline-js-and-tailwind/1852/4).

It's been quite some time since this post and turbo already has `turbo:frame-render` event so maybe this should be included here too?

**Comment by SimoTod** [comment]
Hi, I need to have a look again to remember the issue and see if it applies or it can be fixed now. Feel free to send more details or even a PR if you know already otherwise I'll let you know as soon as I can. Thanks

**Comment by luizkowalski** [comment]
let me try to summarize the problem described in the threads
if you use lazy-loaded turbo frames, like this

```
<%= turbo_frame_tag "tab_content", src: tab_1_content_path %>
```

when the DOM is loaded, this is what Alipine sees

```
<turbo-frame src="..." id="...">

</turbo-frame>
```

so whatever Alpine-enabled componente inside this turbo frame won't be activated because Alpine is like "ok, DOM loaded, I see absolutely nothing here, let's go 🚀"

The proposed solution in the thread is this:

> Upon first page load: set a MutationObserver that will listen to changes inside of all <turbo-frames> present in the first page load. When the interior of those turbo-frames (aka: its child-nodes) change: look for all uninitialized Alpine components, and initialize them.

but the _ideal_ solution would be this one: 

> if Turbo fires an event called something like turbo:frame-loaded every time a frame finishes its rendering after receiving a new request from the server

this thread is a number of months old and today, Turbo does have an event called `turbo:frame-render` that according to the docs, [fires right after a <turbo-frame> element renders its view](https://turbo.hotwired.dev/reference/events). What I don't know yet is if this is triggered after a lazy loaded frame finishes. I will try to confirm this 

edit: just confirmed that `turbo:frame-render` is triggered after a lazy-loaded frame finishes loading so this is the event that the ideal solution is talking about
